aesara/tensor/rewriting/subtensor.py

```diff
@@ -1749,3 +1749,91 @@ def local_join_subtensors(fgraph, node):
                 return [concatenate(new_joined_tensors, axis=axis)]
             else:
                 return [merged_subtensors]
+
+
+@register_specialize
+@node_rewriter(
+    [
+        Subtensor,
+        AdvancedSubtensor1,
+        AdvancedSubtensor,
+        IncSubtensor,
+        AdvancedIncSubtensor,
+        AdvancedIncSubtensor1,
+    ]
+)
+def local_uint_constant_indices(fgraph, node):
+    """Convert constant indices to unsigned dtypes."""
+
+    if isinstance(node.op, (IncSubtensor, AdvancedIncSubtensor, AdvancedIncSubtensor1)):
+        x, y, *indices = node.inputs
+    else:
+        x, *indices = node.inputs
+        y = None
+
+    idx_list = getattr(node.op, "idx_list", None)
+    new_indices = list(indices_from_subtensor(indices, idx_list))
+    has_new_index = False
+
+    for i, index in enumerate(new_indices):
+
+        if not isinstance(index, Constant):
+            continue
+
+        index_val = index.data
+
+        if index_val is None or isinstance(index_val, slice):
+            # TODO: If slice index dtypes matter, we can consider converting
+            # those, as well.
+            continue
+
```

Comment:
Maybe it would be better to add an explicit check for `index_val.dtype == bool`, just to make it obvious what's happening in that case? I think strictly speaking this wouldn't be necessary because a min of booleans still has dtype bool and `np.min_scalar_type(False) == bool`, but it might improve readability and make this case more obvious and also remind anyone working on this code in the future that this is a case that needs to be handles correctly.